go/tools/internal/stdlib_tags/stdlib_tags.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"go/build/constraint"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+// Used to update the list of tags affecting the standard library kept in
+// transitions.bzl.
+func main() {
+	if len(os.Args) < 2 {
+		println("usage: stdlib_tags <go SDK src directory>...")
+		os.Exit(1)
+	}
+
+	tags := make(map[string]struct{})
+	for _, dir := range os.Args[1:] {
+		err := filepath.WalkDir(dir, func(path string, d os.DirEntry, err error) error {
+			if d.IsDir() {
+				if d.Name() == "testdata" {
+					return filepath.SkipDir
+				}
+				return nil
+			}
+			if filepath.Ext(path) != ".go" {
+				return nil
+			}
+			if strings.HasSuffix(filepath.Base(path), "_test.go") {
+				return nil
+			}
+			return walkFile(path, tags)
+		})
+		if err != nil {
+			println(dir + ": " + err.Error())
+			os.Exit(1)
+		}
+	}
+
+	var filteredTags []string
+	for tag := range tags {
+		if !shouldExclude(tag) {
+			filteredTags = append(filteredTags, tag)
+		}
+	}
+	sort.Strings(filteredTags)
+
+	fmt.Printf("_TAG_AFFECTS_STDLIB = {\n")
+	for _, tag := range filteredTags {
+		fmt.Printf("    %q: None,\n", tag)
+	}
+	fmt.Printf("}\n")
+}
+
+var goVersionRegex = regexp.MustCompile(`^go1.(\d+)$`)
```

Comment:
optional nit: I generally prefer to keep top-level variable declarations at the top. This includes the `knownOS` and `knownArch`

Reply:
Moved up the `goVersionRegex`, but I ended up keeping the maps at the end. They are essentially generated code that I think would distract from this otherwise very short tool.